cdap-ui/app/cdap/components/If/index.tsx

```diff
@@ -14,15 +14,34 @@
  * the License.
  */
 import * as React from 'react';
+import Box from '@material-ui/core/Box';
+import withStyles from '@material-ui/core/styles/withStyles';
 
 interface IIfComponentProps {
   condition: boolean;
   children: React.ReactNode;
+  invisible?: boolean;
+  HiddenContainer?: React.ReactNode;
 }
 
-const If: React.FC<IIfComponentProps> = ({ condition, children }) => {
+const If: React.FC<IIfComponentProps> = ({
+  condition,
+  children,
+  invisible = false,
+  HiddenContainer,
+}) => {
   if (!condition) {
-    return null;
+    if (!invisible) {
```

Comment:
this if (!invisible) can be moved outside so that it can be less nested.

Reply:
Hhmm if we flatten it out we ideally check for `invisible` twice but I agree a flat `if` check is easier to read. Have changed it.